### DISCUSSION TITLE
More flexibility for homes GUI

### DISCUSSION THREAD

**Discussion by Tencryn** [Zrips/CMI#7394] [Enhancement]
The homes GUI feels rather limiting, albeit it has a lot of features. I'd like to see a little more customization with the GUI itself. Without going too complex, I have a few option suggestions.

- **Add lore (with placeholders) to the info button**
The reason for this is because I'm wanting to use custom GUIs using fonts, and this takes away the functionality of inventory titles.
As such, being able to have an info button to display the data that would otherwise be in the inventory title (`playerDisplayName`, `current`, and `max`) would be very nice.

- **Disable border when expanded is false**
This would allow other buttons to be placed where the border would've been rather than having a bunch of wasted space. This could be achieved by setting the border to AIR.

- **Change starting row of homes**
While the homes would remain either a 5x2 or 7x4 (expanded true, or with border) and remain centered on the rows, changing the starting row would allow for the homes to be on the first row, rather than being fixed in the middle of a 6 row inventory.

- **Change the amount of rows**
In combination with the above two options, allowing us to change the amount of rows the inventory can have would allow for huge customization. Here are some examples
  * expanded false, 3 rows, buttons at the bottom (or top), homes in the other 2 rows
  * expanded false, 4 rows, buttons on the bottom and the top, homes in the middle 2 rows
  * expanded true, 5 rows, buttons at the bottom (or top), homes in the other 4 rows
  * expanded true, 6 rows, buttons at the bottom and the top, homes in the middle 4 rows

- **Full freedom**
While this is a complex change, changing how many homes are on a page and where the border / background items should be placed would give ultimate flexibility, especially with the ability to change the row count. This would eliminate the need for an expanded option, and allow for 7x2, 9x2, 9x4 and so on for home layouts. I'd be more than happy to see the above changes without the need for this, but this would be nice too (especiallly 7x2)

**Comment by CloudeLecaw** [comment]
i like the displayname and lore idea.. bit at least..this would ment you need a little homes editor....
